ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -1090,3 +1090,22 @@ let requires_passthrough ~__context ~self =
       None
 
 (* Does not require any passthrough *)
+
+let assert_not_legacy ~__context ~self =
+  let legacy =
+    Db.VGPU_type.get_implementation ~__context ~self |> fun x ->
+    match (x, !Xapi_globs.gvt_g_supported) with
+    | `gvt_g, false ->
+        true
+    | _ ->
+        false
+  in
```

Comment:
We want to avoid database queries if possible, since this feature is going to be disabled most of the time I recommend to use short-circuiting to avoid the db: 
```suggestion
  let legacy =
    not !Xapi_globs.gvt_g_supported
    && Db.VGPU_type.get_implementation ~__context ~self = `gvt_g
  in
```